### PR TITLE
Fix warnings coming from ruby -w

### PR DESCRIPTION
**Pull request by zenspider**
This first pass fixes all our circular requires and duplicate method
warnings that come from:

```
ruby -w -Ilib:test test/functional/helper.rb
```

Signed-off-by: Ryan Davis <zenspider@chef.io>

### CODE REVIEW

**Review by miah — APPROVED** [review]

File: lib/inspec/resources/shadow.rb [hunk 1]
```diff
@@ -52,12 +52,6 @@ def initialize(path = "/etc/shadow", opts = {})
       .register_column(:inactive_days, field: "inactive_days")
       .register_column(:expiry_dates, field: "expiry_date")
       .register_column(:reserved, field: "reserved")
-    # These are deprecated, but we need to "alias" them
-    filtertable
-      .register_custom_property(:user) { |table, value| table.resource.user(value) }
-      .register_custom_property(:password) { |table, value| table.resource.password(value) }
-      .register_custom_property(:last_change) { |table, value| table.resource.last_change(value) }
-      .register_custom_property(:expiry_date) { |table, value| table.resource.expiry_date(value) }
```
Comment (clintoncwolfe) [inline lib/inspec/resources/shadow.rb:60]: @james-stocks can we get confirmation it is safe to remove these at this point?

**Review by clintoncwolfe — COMMENTED** [review]
I'm happy with everything, but the `shadow` aliases were there to handle problems we had a while ago. Just need to make sure we don't need them anymore.

File: lib/inspec/resources/shadow.rb (hunk 1 above)
Comment (james-stocks) [reply lib/inspec/resources/shadow.rb:60]: These are what let you directly call like `shadow.user`?

All our supported profiles use `shadow.where` and we don't need anything other than `where`.

Comment (zenspider) [reply lib/inspec/resources/shadow.rb:60]: There were no aliases. There were duplicates. I removed the former, not the latter. All the method names are still there. If I removed names, we'd have test failures (I hope).

**Review by clintoncwolfe — APPROVED** [review]

**Pull request merged**